### PR TITLE
 Call google_set_hostname on openSUSE and when the agent is configured to manage hostname and FQDN, let it.

### PR DESCRIPTION
**Pull request by a-crate**
/cc

### CODE REVIEW

File: src/usr/bin/google_set_hostname [hunk 1]
```diff
@@ -15,6 +15,25 @@
...
+	fi
+	# Don't retain legacy behavior if set_hostname or set_fqdn is enabled in
+	# the agent.
+	if ggactl coreplugin send '{"Command":"agent.config.getoption","Option":"Unstable.SetHostname"}' | grep '"Value":"true"' >/dev/null || ggactl coreplugin send '{"Command":"agent.config.getoption","Option":"Unstable.SetFqdn"}' | grep -i '"Value":"true"' >/dev/null; then
+		return 1
+	fi
+	return 0
+}
+
+if ! retain_legacy_behavior;then
```
Comment (dorileo) [inline src/usr/bin/google_set_hostname:33]: for consistency sake can you add a space between `;` and `then`? :-)

File: src/usr/bin/google_set_hostname [hunk 2]
```diff
@@ -15,6 +15,25 @@
 
 # Deal with a new hostname assignment.
 
+retain_legacy_behavior() {
+	ggactl=$(which ggactl 2> /dev/null)
+	if [ -z "$ggactl" ]; then
+		# Agent doesn't have ggactl for communication, legacy behavior.
+		return 0
+		old_hostname=$($ggactl)
```
Comment (dorileo) [inline src/usr/bin/google_set_hostname:23]: do you really want this `old_hostname` attribution here?

File: src/usr/bin/google_set_hostname [hunk 3]
```diff
@@ -15,6 +15,25 @@
 
 # Deal with a new hostname assignment.
 
+retain_legacy_behavior() {
```
Comment (dorileo) [inline src/usr/bin/google_set_hostname:18]: please add a comment line describing the intended behavior of this routine. I.e what is the definition of legacy behavior and what it means to retain it here.

**Review by dorileo — CHANGES_REQUESTED** [review]

File: src/usr/bin/google_set_hostname (hunk 3 above)
Comment (a-crate) [reply src/usr/bin/google_set_hostname:24]: Done.

File: src/usr/bin/google_set_hostname (hunk 2 above)
Comment (a-crate) [reply src/usr/bin/google_set_hostname]: No, it's not even reachable.

File: src/usr/bin/google_set_hostname (hunk 1 above)
Comment (a-crate) [reply src/usr/bin/google_set_hostname]: Done.

**Pull request merged**